deps/rabbit/src/rabbit_classic_queue.erl

```diff
@@ -63,14 +64,61 @@
          send_drained_credit_api_v1/4,
          send_credit_reply/7]).
 
+-export([validate_policy/1]).
+
+-rabbit_boot_step(
+   {?MODULE,
+    [{description, "Deprecated queue-master-locator support."
+      "Use queue-leader-locator instead."},
+     {mfa, {rabbit_registry, register,
+            [policy_validator, <<"queue-master-locator">>, ?MODULE]}},
+     {mfa, {rabbit_registry, register,
+            [operator_policy_validator, <<"queue-master-locator">>, ?MODULE]}},
+     {requires, rabbit_registry},
+     {enables, recovery}]}).
+
+validate_policy(Args) ->
+    Strategy = proplists:get_value(<<"queue-master-locator">>, Args, unknown),
+    case rabbit_queue_location:master_locator_permitted() of
+        true ->
+            case lists:member(Strategy, rabbit_queue_location:queue_leader_locators()) of
+                true -> ok;
+                false -> {error, "~tp is not a valid master locator", [Strategy]}
+            end;
+        false ->
+            {error, "use of deprecated queue-master-locator argument is not permitted", []}
```

Comment:
Should we say here `policy` instead of `argument` given that this validates a policy and not a queue argument?